R/harmonic.R

```diff
@@ -0,0 +1,374 @@
+#' Add sin and cos terms for harmonic analysis
+#'
+#' `step_harmonic` creates a *specification* of a recipe step that
+#'   will add sin and cos terms for harmonic analysis.
+#'
+#' @inheritParams step_center
+#'
+#' @param recipe A recipe object. The step will be added to the sequence of
+#'   operations for this recipe.
+#' @param ... One or more selector functions to choose which variables are
+#'  affected by the step. See [selections()] for more details.
+#' @param role For model terms created by this step, what analysis
+#'  role should they be assigned?. By default, the function assumes
+#'  that the new columns created from the original variables will be
+#'  used as predictors in a model.
+#' @param trained A logical to indicate if the quantities for preprocessing
+#'   have been estimated. Again included for consistency.
+#' @param frequency A numeric vector with at least one value if using frequency
+#'   to specify the cycle size.  The value must be greater than zero and finite.
+#' @param period A numeric vector with at least one value if using period to
+#'   specify the cycle size. The value must be greater than zero and finite.
+#' @param cycle_unit Character string to indicate the units for a cycle.
+#'   Possible values are:
+#'   'year' (31556926 seconds), 'month_synodic' (2360592 seconds),
+#'   'month_sidereal' (2551443 seconds), 'month_average' (2629744 seconds),
+#'   'week', day', 'hour', 'minute', 'second', 'sample'.
+#' @param starting_val either numeric, Date or POSIXt value that indicates the
+#'   reference point for the sin and cos curves.  This parameter may be
+#'   specified to increase control over the signal phase.  If starting_val is
+#'   not specified the default is the start of the data series, or the start of
+#'   the cycle (i.e. the beginning of the hour, day, week).
+#' @param objects Statistics are stored here once this step has
+#'  been trained by [prep.recipe()].
+#'
```

Comment:
We'll need a Details section where you explain what this is doing and where it would be appropriate to use.

Reply:
A reference would be good too.

Reply:
Added details with a couple of references and another example to provide a little more background.